portal-bridge/src/types/state/walk_trie_diff.rs

```diff
@@ -0,0 +1,101 @@
+use alloy_primitives::{Bytes, B256};
+use eth_trie::{decode_node, node::Node};
+use hashbrown::HashMap as BrownHashMap;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct AccountProof {
+    pub path: Vec<u8>,
+    pub proof: Vec<Bytes>,
+}
+
+/// This struct takes in a root hash and a hashmap of changed nodes, then you can call an iterator
+/// which will return every proof to gossip
+pub struct WalkTrieDiff {
+    pub changed_nodes: BrownHashMap<Vec<u8>, Vec<u8>>,
+    path_backwards: BrownHashMap<Vec<u8>, (Vec<u8>, u8)>,
```

Comment:
> if you want more optimized version

I am currently deleting all the in memory database code in my next PR. I don't think it is the time to optimize stuff when it can and will be all rewritten over and over none of this code is stable

Reply:
I will call it PathToParent, as it isn't ParentInfo